types/decimal.go

```diff
@@ -138,7 +138,7 @@ func NewDecFromIntWithPrec(i Int, prec int64) Dec {
 // CONTRACT - This function does not mutate the input str.
 func NewDecFromStr(str string) (Dec, error) {
 	if len(str) == 0 {
-		return Dec{}, ErrEmptyDecimalStr
+		return Dec{}, fmt.Errorf("%w: %s", ErrEmptyDecimalStr, str)
```

Comment:
I think this should be the other way around: `fmt.Errorf("%s: %w", str, ErrEmptyDecimalStr)